registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java

```diff
@@ -268,19 +271,25 @@ public MessageDTO process(MessageDTO object) {
 
 					String test = (String) registrationProcessorRestClientService.getApi(ApiName.UINGENERATOR, null, "",
 							"", String.class);
-
 					regProcLogger.info(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
 							registrationId, "Received response from UINGENERATOR API");
-
 					Gson gsonObj = new Gson();
 					uinResponseDto = gsonObj.fromJson(test, UinGenResponseDto.class);
-
 					uinField = uinResponseDto.getResponse().getUin();
 					demographicIdentity.put("UIN", uinField);
-
 					idResponseDTO = sendIdRepoWithUin(registrationId, registrationStatusDto.getRegistrationType(), demographicIdentity,
 							uinField, description);
+					if(idResponseDTO.getErrors()!=null && idResponseDTO.getErrors().get(0).getErrorCode().equalsIgnoreCase(INVALID_INPUT_PARAMETER_ERROR_CODE)) {
 
+						for (int i = registrationStatusDto.getRetryCount(); i < maxRetrycount; i++) {
+
+								idResponseDTO = sendIdRepoWithUin(registrationId, registrationStatusDto.getRegistrationType(), demographicIdentity,
+										uinField, description);
+								if (idResponseDTO.getErrors()==null || idResponseDTO.getErrors().get(0).getErrorCode()!=INVALID_INPUT_PARAMETER_ERROR_CODE)
+									break;
+								registrationStatusDto.setRetryCount(registrationStatusDto.getRetryCount() + 1);

```

Comment:
registrationStatusDto.setRetryCount and retry we are dealing 2 different retries, lets no tie them both.
If we do this second time the reprocessor picks up it might not try 3 times again, rather it will try only once

Reply:
Done